kubernetes/adapter.go

```diff
@@ -217,7 +211,29 @@ func (a *Adapter) parseAnnotations(annotations map[string]string) *Ingress {
 		sslPolicy = a.ingressDefaultSSLPolicy
 	}
 
-	loadBalancerType := getAnnotationsString(annotations, ingressLoadBalancerTypeAnnotation, a.ingressDefaultLoadBalancerType)
+	loadBalancerType, hasLB := annotations[ingressLoadBalancerTypeAnnotation]
+	if !hasLB {
+		loadBalancerType = a.ingressDefaultLoadBalancerType
+	}
+
+	securityGroup, hasSG := annotations[ingressSecurityGroupAnnotation]
+	if !hasSG {
+		securityGroup = a.ingressDefaultSecurityGroup
+	}
+
+	wafWebAclId, hasWAF := annotations[ingressWAFWebACLIDAnnotation]
+
+	if (loadBalancerType == loadBalancerTypeNLB) && (hasSG || hasWAF) {
+		log := log.WithFields(log.Fields{"type": typ, "ns": metadata.Namespace, "name": metadata.Name})
+
+		if hasLB {
+			log.Info("Security Group or WAF are not supported by NLB (configured by annotation)")
```

Comment:
It seems to me like this should be an error and we should not create/update the ingress in this case but ignore it. Otherwise it's maybe confusing for the user that it was configured but not with the intended settings.

(ofc. we could also argue that this could be caught at admission/deployment, before the ingress resource is even submitted to the cluster).

Reply:
@mikkeloscar I have to disagree.
Imagine the situation is like this:
You have a WAF and default to ALB. Later you switch default to NLB, which will do nothing and stay at ALB.
Now you have an incident that can be mitigated by NLB. Hurry up and you set the lb annotation to specify "I WANT AN NLB!!!!". This is what I imagine in this case, so I think it is right to use NLB in this case.

Reply:
After discussions we agreed on to follow @mikkeloscar 's idea. Errors in configuration will be seen only in logs.